src/main.cpp

```diff
@@ -866,12 +928,59 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
 {
     // Basic checks that don't depend on any context
 
-    // Check transaction version
-    if (tx.nVersion < MIN_TX_VERSION) {
+    // Pre-Overwinter accepts version >=1 transaction. Overwinter only accepts version 3 transactions.
+    // To enable soft-forks of the transaction format, define a range of versions which can be accepted.
+    if (tx.fOverwintered) {
+        if (tx.nVersion < OVERWINTER_MIN_TX_VERSION) {
+            return state.DoS(100, error("CheckTransaction(): overwinter version too low"),
+                REJECT_INVALID, "bad-tx-overwinter-version-too-low");
+        }
+        else if (tx.nVersion < CTransaction::OVERWINTER_MIN_CURRENT_VERSION) {
+            return state.DoS(100, error("CheckTransaction(): overwinter version too low for client"),
+                REJECT_INVALID, "bad-tx-overwinter-version-client-too-low");
+        }
+        else if (tx.nVersion > CTransaction::OVERWINTER_MAX_CURRENT_VERSION ) {
```

Comment:
Define and use an `OVERWINTER_MAX_TX_VERSION` (and don't include `OVERWINTER_MAX_CURRENT_VERSION` here for the same reason as above).